### PR TITLE
BUG: Fix bounds checking for random.logseries

### PR DESCRIPTION
**Pull request by seberg**
Logseries previously did not enforce bounds to be strictly exclusive. This leads to incorrect behavior for 0 and 1.  Add a new path to implement the correct bounds-checking.

The NOT_NAN check is removed, since it was never used: The current bounded version always excludes NaNs.

### CODE REVIEW

**Review by WarrenWeckesser — APPROVED** [review]
Looks good to me, let's see if @bashtage has any suggestions.

**Review by bashtage — CHANGES_REQUESTED** [review]
I think the mtrand changes are outside of the allowed scope per NEP19.

**Comment by bashtage** [comment]
Does the distribution behave sensibly in a neighborhood of 0+?  Or is 0 a special case and other tiny numbers are less reliable?

**Comment by seberg** [comment]
I have not spelled it out, but the path for 1 hangs, so 1 was never a valid input.  Changing 0 to be excluded would indeed be a change that may be covered by the NEP.
Other tiny numbers work just as well (they always return 1 immediately), I am not sure if there is a loss of precision close to 1, but in that case I still guess this is right and that would be a different enhancement.

File: numpy/random/mtrand.pyx [hunk 1]
```diff
@@ -3966,7 +3966,7 @@ cdef class RandomState:
         Parameters
         ----------
         p : float or array_like of floats
-            Shape parameter for the distribution.  Must be in the range (0, 1).
+            Shape parameter for the distribution.  Must be in the range [0, 1).
```
[the remark this answers is not in the document]
Comment (charris) [reply numpy/random/mtrand.pyx:3969]: Yes, that seems like a reasonable take.

[the remark this answers is not in the document]
Comment (seberg) [reply numpy/random/mtrand.pyx:3969]: This is only a doc update?  There is no change in behavior.  0 always worked, as pointed out by Warren, 1 never worked it is just explicit now.

**Comment by charris** [comment]
Thanks Sebastian.

**Pull request merged**